### DISCUSSION TITLE
Remember me function forget the user after the session expiration

### DISCUSSION THREAD

**Discussion by AlbertoMoricca** [codeigniter4/shield#533] [Q&A] [answered]
Hi,
I'm using Codeigniter 4 with Shield framework. I set the session duration to 5 minutes for test scope, and the remember me function is enabled in the Auth.php (allowRemembering = true). 

Furthermore, if remember me checkbox is set to true on login by the user, I call the function `$loginAttempt = auth()->remember()->attempt($credentials);` instead `$loginAttempt = auth()->attempt($credentials);`

But after 5 minutes of inactivity, the function `auth()->loggedIn();` return nothing (instead of 1, like happens before 5 minutes). What's the issue?

Thanks a lot,
Alberto

**Comment by kenjis** [comment, marked as answer]
Do you have the remember-me cookie in your browser?

**Reply by AlbertoMoricca** [reply]
Hi,

How can i check this? The other website Remember me function works fine.

Furthermore, I have the token correctly saved on auth_remember_token DB table (1 month duration)

**Reply by kenjis** [reply]
You can see cookies the developer tool in your browser.

**Reply by AlbertoMoricca** [reply]
I checked the browser and i found 1 Cookie of the site before and after the login with the remember me enabled. I have no details about this cookie, then it's not clear if it's a remember me cookie or other type of cookie. Suppose that the cookie is not a remember me cookie. I tried on Chrome and Firefox and the issue is the same. I don't think that is a browser issue. Do you have any ideas for another check I can do?

**Reply by AlbertoMoricca** [reply]
I found the solution: after the login with the remember me, I perform a redirect. It seems that immediately after setting a cookie, the redirect need the "->withCookies();" istruction to avoid cookie lost. Now the remember me works. Thanks for the help.
